### PR TITLE
feat: adding gin-redis sample app documentation

### PR DESCRIPTION
**Pull request by Shashwat79802**
# Pull Request Template

## Description

Please include a summary of the change. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)

...

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added corresponding tests.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: versioned_docs/version-2.0.0/quickstart/go-gin-redis.md [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+  ```
+
+  ### Lights, Camera, Record! 🎥
+
+  Capture the test-cases-
+  ```shell
+  keploy record -c "docker run -p 3001:3001 --name RediApp --network <networkName> --name ginRedisApp gin-app:1.0"
+  ```
+
+ 🔥**Make some API calls**. Postman, Hoppscotch or even curl - take your pick!
```
Comment (Sonichigo) [inline versioned_docs/version-2.0.0/quickstart/go-gin-redis.md:474]: This is showing out the details and summary part

Comment (Shashwat79802) [reply versioned_docs/version-2.0.0/quickstart/go-gin-redis.md]: Correction made!!

File: versioned_docs/version-2.0.0/quickstart/go-gin-redis.md [hunk 2]
```diff
@@ -0,0 +1,638 @@
...
+  Start the Redis instance using the `docker-compose` file-
+
+  ```bash
+  sudo docker run -p 6379:6379 -d --network <networkName> --name myredis redis
+  ```
+
+  ```bash
+  docker build -t gin-app:1.0 .
+  ```
+
```
Comment (Sonichigo) [inline versioned_docs/version-2.0.0/quickstart/go-gin-redis.md:45]: I think this section will be after the cloning the repository and cd gin-redis

**Review by Sonichigo — CHANGES_REQUESTED** [review]

File: versioned_docs/version-2.0.0/quickstart/go-gin-redis.md (hunk 2 above)
Comment (Shashwat79802) [reply versioned_docs/version-2.0.0/quickstart/go-gin-redis.md:52]: Done!!

**Review by Sonichigo — APPROVED** [review]
Looks good to me 🚀

**Pull request merged**